app/assets/javascripts/govuk_publishing_components/components/search-with-autocomplete.js

```diff
@@ -0,0 +1,99 @@
+/* global accessibleAutocomplete, fetch */
+//= require accessible-autocomplete/dist/accessible-autocomplete.min.js
+
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  class SearchWithAutocomplete {
+    constructor ($module) {
+      this.$module = $module
+
+      this.$originalInput = this.$module.querySelector('input')
+      this.$inputWrapper = this.$module.querySelector('.gem-c-search__input-wrapper')
+      this.$form = this.$module.closest('form')
+
+      this.source = this.$module.getAttribute('data-source')
+      this.sourceKey = this.$module.getAttribute('data-source-key')
+    }
+
+    init () {
+      // Do not enhance the input field if we won't be able to provide autocomplete suggestions
+      if (!this.source || !this.sourceKey) {
+        console.warn('search-with-autocomplete: No source/sourceKey provided')
+        return
+      }
```

Comment:
Is this necessary if we're raising in the component? I'd imagine the only way you could get this is an unsupported use case of using the component JS without the HTML.

If we do need something we may want something a bit more aggressive given the unsupported nature of it.

Reply:
I've added it in for just-in-case defensiveness, but given it's in Publishing Components and not meant to be used separately, agree it can go.